### PR TITLE
Feat assignment 9 memory management 2

### PR DESCRIPTION
**Pull request by aristovinceselvanathan**

### CODE REVIEW

File: src/Assignment9/ValueAndReferenceTypes/Program.cs [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+        /// <summary>
+        /// It updates the variable by value type
+        /// </summary>
+        /// <param name="value">It takes the value as integer to change itr</param>
+        public static void UpdateValue(int value)
+        {
+            value = 200;
+            Console.WriteLine($"Value in the function: {value}");
+        }
```
Comment (sandeep-sol) [inline src/Assignment9/ValueAndReferenceTypes/Program.cs]: The task asks to create only one function.

Having two functions will be more meaningful. But is it intentional?

File: src/Assignment9/ValueAndReferenceTypes/Program.cs [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            Console.WriteLine("Press Any Key to Continue:");
+            Console.ReadKey();
+            Console.WriteLine("To create the Large Array of Integer");
+            LargeArray();
+            Console.WriteLine("Press Any Key to Continue:");
+            Console.ReadKey();
+        }
+
+        /// <summary>
+        /// It will change the value by reference
```
Comment (sandeep-sol) [inline src/Assignment9/ValueAndReferenceTypes/Program.cs]: Please change the format of this description.

Applicable for the other function and param descriptions.

Hope not to repeat this comment after this occurrence.

File: src/Assignment9/ValueAndReferenceTypes/Program.cs [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        /// <summary>
+        /// Main method that creates the Value and Reference Type objects
+        /// </summary>
+        /// <param name="args">It takes the string array from the command line interface</param>
+        public static void Main(string[] args)
+        {
+            Employee employee = new Employee();
+            int value = 100;
+
+            employee.Id = "1001";
```
Comment (sandeep-sol) [inline src/Assignment9/ValueAndReferenceTypes/Program.cs]: Please update the constructor of the `Employee` class to have these values by default.

File: src/Assignment9/ValueAndReferenceTypes/Program.cs [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        /// </summary>
+        /// <param name="value">It takes the value as integer to change itr</param>
+        public static void UpdateValue(int value)
+        {
+            value = 200;
+            Console.WriteLine($"Value in the function: {value}");
+        }
+
+        /// <summary>
+        /// Method creates the large array of Integers.
```
Comment (sandeep-sol) [inline src/Assignment9/ValueAndReferenceTypes/Program.cs]: The requirement is to extend the project and use it. Not to use the same project.

Please update it.

File: src/Assignment9/Readme.md [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+### Task Steps:
+
+1. Create a new C# Console Application named "IDisposableDemo".
+2. Create a class that opens a file for writing and implements the IDisposable interface. In the Dispose method, ensure that the file is properly closed and released.
+3. In the Main method, create an instance of this class in a 'using' block. Write some text to the file.
+4. After the 'using' block, attempt to open the same file for reading. If the file was properly released, this should be possible without any errors.<br>
+**Expected Outcome**: The application should be able to open the file for reading after the 'using' block,
+demonstrating that the file was properly released when the IDisposable object was disposed.
+
+# Documentation Report:
```
Comment (sandeep-sol) [inline src/Assignment9/Readme.md:59]: Please add your understanding of each topic/task in detail in this report.

**Review by sandeep-sol — CHANGES_REQUESTED** [review]

File: src/Assignment9/ValueAndReferenceTypes/Program.cs (hunk 1 above)
Comment (aristovinceselvanathan) [reply src/Assignment9/ValueAndReferenceTypes/Program.cs]: I have updated the logic

File: src/Assignment9/ValueAndReferenceTypes/Program.cs (hunk 2 above)
Comment (aristovinceselvanathan) [reply src/Assignment9/ValueAndReferenceTypes/Program.cs]: I have changed it

File: src/Assignment9/ValueAndReferenceTypes/Program.cs (hunk 3 above)
Comment (aristovinceselvanathan) [reply src/Assignment9/ValueAndReferenceTypes/Program.cs]: I have declared it in constructor itself.

File: src/Assignment9/ValueAndReferenceTypes/Program.cs (hunk 4 above)
Comment (aristovinceselvanathan) [reply src/Assignment9/ValueAndReferenceTypes/Program.cs]: I have extended to another project of Name ExtendedValueAndReferenceType.

File: src/Assignment9/Readme.md (hunk 5 above)
Comment (aristovinceselvanathan) [reply src/Assignment9/Readme.md:59]: I have added the understandings of the each topic

File: src/Assignment9/ValueAndReferenceTypes/Employee.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Employee"/> class.
+        /// It sets the predefined value to the Employee object.
+        /// </summary>
+        public Employee()
+        {
+            this.Id = "1001";
+            this.Name = "Tom Cruise";
+            this.Designation = "IT";
+        }
```
Comment (sandeep-sol) [inline src/Assignment9/ValueAndReferenceTypes/Employee.cs:19]: Generally, we should use a parameterized constructor for these cases. Please update this.

File: src/Assignment9/ValueAndReferenceTypes/Employee.cs [hunk 7]
```diff
@@ -0,0 +1,52 @@
+﻿namespace ValueAndReferenceTypes
+{
+    /// <summary>
+    /// Employee Class
+    /// </summary>
+    internal class Employee
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Employee"/> class.
+        /// It sets the predefined value to the Employee object.
```
Comment (sandeep-sol) [inline src/Assignment9/ValueAndReferenceTypes/Employee.cs]: Please remove this statement. It is not necessary.

File: src/Assignment9/ValueAndReferenceTypes/Program.cs [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+    /// <summary>
+    /// Program Class that contains the entry of the program
+    /// </summary>
+    internal class Program
+    {
+        /// <summary>
+        /// Main method that creates the Value and Reference Type field and print the fields
+        /// </summary>
+        /// <param name="args">It takes the string array from the command line interface</param>
```
Comment (sandeep-sol) [inline src/Assignment9/ValueAndReferenceTypes/Program.cs]: I've already mentioned updating these types of param descriptions. Please update it in all the places.

File: src/Assignment9/ValueAndReferenceTypes/Program.cs [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+        public static void Main(string[] args)
+        {
+            Employee employee = new Employee();
+            int value = 100;
+
+            Console.WriteLine("Reference Type : Employee System");
+            Console.WriteLine($"Value is {value}");
+            Console.WriteLine(employee);
```
Comment (sandeep-sol) [inline src/Assignment9/ValueAndReferenceTypes/Program.cs:20]: This is not meaningful. Are these lines interchanged?

File: src/Assignment9/ValueAndReferenceTypes/Program.cs [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+            ChangeValues(employee, value);
+            Console.WriteLine("After the function call:");
+            Console.WriteLine(employee);
+            Console.WriteLine($"Value is after calling the function to change value is {value}\n");
```
Comment (sandeep-sol) [inline src/Assignment9/ValueAndReferenceTypes/Program.cs]: Typos.. Please update it...

File: src/Assignment9/Readme.md [hunk 11]
```diff
@@ -0,0 +1,133 @@
...
+
+1. Generation 0: The generation 0 of the heap memory contains all the transient objects, such as temporary variables. Except for big objects, all newly allocated objects implicitly belong to generation 0. In general, generation 0 has the highest frequency of waste collection.
+
+2. Generation 1: If some generation 0 objects take up space and are not removed during a trash collection cycle, such things are transferred to generation 1. The objects in this generation act as a kind of buffer between the generation 0 objects that have a short lifespan and the generation 2 things that have a long lifespan.
+
+3. Generation 2: If certain generation 1 objects take up space that isn't freed up in the following trash collection cycle, those things are transferred to generation 2. The generation 2 objects have a lengthy lifespan since they stay in the heap memory throughout the entire process, like static objects.
+
+## Understandings of the Project
```
Comment (sandeep-sol) [inline src/Assignment9/Readme.md:85]: Expecting a detailed explanation of all of these.

File: src/Assignment9/ValueAndReferenceTypes/Program.cs [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+            Console.WriteLine("To create the Large Number of Local Variable");
+            ExtendedValueAndReferenceType.LargeNumberLocalVariable();
+            Console.WriteLine("Press Any Key to Continue:");
+            Console.ReadKey();
+            Console.WriteLine("To create the Large Array of Integer");
+            ExtendedValueAndReferenceType.LargeArray();
```
Comment (sandeep-sol) [inline src/Assignment9/ValueAndReferenceTypes/Program.cs]: Please share your observations from these 2 function calls in this comment thread.

**Review by sandeep-sol — CHANGES_REQUESTED** [review]

File: src/Assignment9/ValueAndReferenceTypes/Employee.cs (hunk 6 above)
Comment (aristovinceselvanathan) [reply src/Assignment9/ValueAndReferenceTypes/Employee.cs:19]: I have created the constructor with parameters.

File: src/Assignment9/ValueAndReferenceTypes/Employee.cs (hunk 7 above)
Comment (aristovinceselvanathan) [reply src/Assignment9/ValueAndReferenceTypes/Employee.cs]: I have removed the comment line

File: src/Assignment9/ValueAndReferenceTypes/Program.cs (hunk 8 above)
Comment (aristovinceselvanathan) [reply src/Assignment9/ValueAndReferenceTypes/Program.cs]: I have changed it.

File: src/Assignment9/ValueAndReferenceTypes/Program.cs (hunk 9 above)
Comment (aristovinceselvanathan) [reply src/Assignment9/ValueAndReferenceTypes/Program.cs:20]: I have updated it.

File: src/Assignment9/ValueAndReferenceTypes/Program.cs (hunk 10 above)
Comment (aristovinceselvanathan) [reply src/Assignment9/ValueAndReferenceTypes/Program.cs]: I have updated it.

File: src/Assignment9/ValueAndReferenceTypes/Program.cs (hunk 12 above)
Comment (aristovinceselvanathan) [reply src/Assignment9/ValueAndReferenceTypes/Program.cs]: I have commented on that line about the functionality of functions

File: src/Assignment9/Readme.md (hunk 11 above)
Comment (aristovinceselvanathan) [reply src/Assignment9/Readme.md:85]: I have added it

**Review by sandeep-sol — APPROVED** [review]

**Pull request merged**